src/read.c

```diff
@@ -4257,21 +4267,38 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
             maxItemID = item->id;
         }
         if (item->dimgForID == colorItem->id) {
+            avifBool seenAlphaForCurrentItem = AVIF_FALSE;
             for (uint32_t j = 0; j < meta->items.count; ++j) {
                 avifDecoderItem * auxlItem = &meta->items.item[j];
                 if (avifDecoderItemIsAlphaAux(auxlItem, item->id)) {
+                    if (seenAlphaForCurrentItem) {
+                        // Multiple items are claiming to be the alpha auxiliary of the current item.
+                        avifFree(alphaItemIndices);
+                        *alphaItem = NULL;
+                        *isAlphaItemInInput = AVIF_FALSE;
+                        return AVIF_RESULT_INVALID_IMAGE_GRID;
+                    }
+                    if (auxlItem->dimgForID != 0) {
+                        // Alpha auxiliary is dimg for another item. This is invalid.
+                        avifFree(alphaItemIndices);
+                        *alphaItem = NULL;
+                        *isAlphaItemInInput = AVIF_FALSE;
+                        return AVIF_RESULT_INVALID_IMAGE_GRID;
+                    }
```

Comment:
Nit: merge these two if statements (lines 4274-4287)?

Note: I am tempted to suggest we use goto statements to share the cleanup code. But let's be conservative and focus on the bug fix in this pull request.

Reply:
> Nit: merge these two if statements (lines 4274-4287)?
> 

Done.

> Note: I am tempted to suggest we use goto statements to share the cleanup code. But let's be conservative and focus on the bug fix in this pull request.

Yes, i was planning on doing this in a follow-up anyway.